src/crimson/os/alienstore/alien_store.h

```diff
@@ -124,5 +174,6 @@ class AlienStore final : public FuturizedStore {
   std::unique_ptr<CephContext> cct;
   mutable seastar::gate op_gate;
   std::unordered_map<coll_t, CollectionRef> coll_map;
+  seastar::sharded<AlienShardStore> shard_stores;
```

Comment:
It is not needed, there should be no internal changes to AlienStore.

The AlienStore can inherit from both FuturizedStore and FuturizedShardStore, with no special internal changes to work exactly as before. This is also why the interfaces from FuturizedStore and FuturizedShardStore should be disjoint.